legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -212,3 +213,171 @@ def search_businesses():
+        mailing_address.delivery_instructions = 'HJ Test' if not mailing_address.delivery_instructions else ''
+    else:
+        print(f'Skip updating mailing address for Business(id={business.id})')
+    business.save()
+    business = Business.find_by_identifier(identifier)
+    return jsonify({'success': True, 'id': business.id, 'admin_freeze': business.admin_freeze, 'address_id': mailing_address.id, 'delivery_instructions': mailing_address.delivery_instructions}), HTTPStatus.OK
+
+
+@bp.route('/db_versioning_test/insert', methods=['GET'])
+@cross_origin(origins='*')
+def db_versioning_test_insert():
+    """Return a JSON object with meta information about the Service.
+
+    Create and insert a dummy business.
+    """
+    identifier = 'BC0000001'
+    business = Business(identifier=identifier, legal_name='HJ Test db versioning')
+    business.save()
+    return jsonify({'success': True, 'id': business.id})
+
+
+@bp.route('/db_versioning_test/inserts', methods=['GET'])
+@cross_origin(origins='*')
+def db_versioning_test_inserts():
+    """Return a JSON object with meta information about the Service.
+
+    Create and insert a dummy business with FK related offices, addresses.
+    """
+    identifier = 'BC0000001'
+    business = Business(identifier=identifier, legal_name='HJ Test db versioning')
+    address=Address(
+        city='HJ TEST City',
+        street='HJ Test Street',
+        postal_code='X1X1X1',
+        country='CA',
+        region='BC',
+        address_type=Address.MAILING
+    )
+    office=Office(office_type='registeredOffice')
+    office.addresses.append(address)
+    business.offices.append(office)
+    business.save()
+    return jsonify({'success': True, 'business_id': business.id, 'office_id': office.id, 'address_id': address.id})
+
+
+@bp.route('/db_versioning_test/delete/<int:business_id>', methods=['GET'])
+@cross_origin(origins='*')
+def db_versioning_test_delete(business_id):
+    """Return a JSON object with meta information about the Service.
+
+    Delete a dummy business.
+    Need a dummy business without FK related records to test this endpoint.
+    """
+    business = Business.find_by_internal_id(business_id)
+    db.session.delete(business)
+    db.session.commit()
+    return jsonify({'success': True, 'id': business_id})
+
+
+@bp.route('/db_versioning_test/deletes/<int:business_id>', methods=['GET'])
+@cross_origin(origins='*')
+def db_versioning_test_deletes(business_id):
+    """Return a JSON object with meta information about the Service.
+
+    Delete a dummy business with FK related offices, addresses.
+    Need a dummy business with FK related records to test this endpoint.
+    """
+    business = Business.find_by_internal_id(business_id)
+    db.session.delete(business)
+    db.session.commit()
+    return jsonify({'success': True, 'id': business_id})
+
+
+@bp.route('/db_versioning_test/flush', methods=['GET'])
+@cross_origin(origins='*')
+def db_versioning_test_flush():
```

Comment:
This is an extra scenario to test several `add` and `flush` operations before final `commit`. The multiple `add` and `flush` operations are all in one single transaction and should not result in multiple transaction records.

Reply:
thanks for the thorough testing